src/Universe.cpp

```diff
@@ -1720,3 +1563,228 @@ int Lattice::getLatticeSurface(int cell, Point* point) {
 
   return surface;
 }
+
+
+/**
+  * @brief  Calculates the boundary locations and 
+  *         conditions (VACUUM or REFLECTIVE) at the
+  *         maximum and minimum reachable coordinates in the Universe
+  */
+void Universe::calculateBoundaries() {
+
+  /* Calculate the minimum reachable x-coordinate in the geometry and store it
```

Comment:
Can we try to simplify this function so that we only loop over cells once and just figure out what to do on each cell?